lib/blueprinter/helpers/base_helpers.rb

```diff
@@ -17,12 +19,28 @@ def prepare_for_render(object, options)
         prepare(object, view_name: view_name, local_options: options, root: root, meta: meta)
       end
 
+      def self.set_should_preload_associations(should_preload_associations)
+        @@should_preload_associations = should_preload_associations
+      end
+
+      def preload_associations?
+        if @@should_preload_associations.nil? || @@should_preload_associations == false
+          false
+        else
+          true
+        end
+      end
+
       def prepare_data(object, view_name, local_options)
         if array_like?(object)
-          object.map do |obj|
-            object_to_hash(obj,
-                           view_name: view_name,
-                           local_options: local_options)
+          if object.respond_to?(:reflect_on_all_associations) && preload_associations?
```

Comment:
What happens if the object given is already eager loaded?

Reply:
Good call. I'll add a check for this